debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/OracleStreamingChangeEventSource.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.oracle.logminer;
+
+import io.debezium.connector.oracle.OracleConnectorConfig;
+import io.debezium.connector.oracle.OracleDatabaseSchema;
+import io.debezium.connector.oracle.OracleOffsetContext;
+import io.debezium.jdbc.JdbcConnection;
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.source.spi.StreamingChangeEventSource;
+import io.debezium.relational.TableId;
+import io.debezium.util.Clock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * A {@link StreamingChangeEventSource} based on Oracle's LogMiner utility.
+ * The event handler loop is executed in a separate executor. todo:  implement
+ */
+public class OracleStreamingChangeEventSource implements StreamingChangeEventSource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleStreamingChangeEventSource.class);
+
+    private final JdbcConnection jdbcConnection;
+    private final EventDispatcher<TableId> dispatcher;
+    private final ErrorHandler errorHandler;
+    private final Clock clock;
+    private final OracleDatabaseSchema schema;
+    private final OracleOffsetContext offsetContext;
+    private final boolean tablenameCaseInsensitive;
+    //private final int posVersion;
+
+    public OracleStreamingChangeEventSource(OracleConnectorConfig connectorConfig, OracleOffsetContext offsetContext, JdbcConnection jdbcConnection, EventDispatcher<TableId> dispatcher, ErrorHandler errorHandler, Clock clock, OracleDatabaseSchema schema) {
+        this.jdbcConnection = jdbcConnection;
+        this.dispatcher = dispatcher;
+        this.errorHandler = errorHandler;
+        this.clock = clock;
+        this.schema = schema;
+        this.offsetContext = offsetContext;
+        this.tablenameCaseInsensitive = connectorConfig.getTablenameCaseInsensitive();
+        //this.posVersion = connectorConfig.getOracleVersion().getPosVersion();
+    }
+
+    @Override
+    public void execute(ChangeEventSourceContext context) throws InterruptedException {
```

Comment:
Ok, that answers my question about the actual mining loop :) There were a few interesting comments on the [DBZ-137](https://issues.jboss.org/browse/DBZ-585) issue not too long ago. Definitely worth checking out.

Reply:
I reviewed latest comments and still think LogMiner should be the best option for now.
In defending of it I could say XStream seems to use it under the hood.
Also other commercial companies like Striim use it. BTW one of the Striim founders is GoldenGate designer.

Reply:
Ok, cool. I think it will all be clearer once the actual polling loop is here.